pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1542,23 +1112,7 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
 		// completed, either successfully or with a failure.
 		if resourceClaimModified {
 			if isExtendedResourceClaim {
-				// Unlike other claims, extended resource claim is created in API server below.
-				// AssumeClaimAfterAPICall returns ErrNotFound when the informer update has not reached assumed cache yet.
-				// Hence we must poll and wait for it.
-				pollErr := wait.PollUntilContextTimeout(ctx, 1*time.Second, time.Duration(AssumeExtendedResourceTimeoutDefaultSeconds)*time.Second, true,
-					func(ctx context.Context) (bool, error) {
-						if err := pl.draManager.ResourceClaims().AssumeClaimAfterAPICall(claim); err != nil {
-							if errors.Is(err, assumecache.ErrNotFound) {
-								return false, nil
-							}
-							logger.V(5).Info("Claim not stored in assume cache", "claim", klog.KObj(claim), "err", err)
-							return false, err
-						}
-						return true, nil
-					})
-				if pollErr != nil {
-					logger.V(5).Info("Claim not stored in assume cache after retries", "claim", klog.KObj(claim), "err", pollErr)
-				}
+				pl.waitForExtendedClaimInAssumeCache(ctx, logger, claim)
```

Comment:
Picking just one example: having a call to a method with a descriptive name instead of a wall of source code improves readability. The method itself that has a separate function comment explaining what it does, which also makes it is easier to understand.